### PR TITLE
Feat/add axios retry and better errors

### PR DESCRIPTION
**Pull request by DmytroHryshyn**

### CODE REVIEW

File: src/axios/index.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import axios from 'axios';
+import axiosRetry from 'axios-retry';
+
+const client = axios.create();
+axiosRetry(client);
+
+export const DEFAULT_RETRY_COUNT = 3;
+export default client;
```
Comment (grzpab) [inline src/axios/index.ts:8]: can we call it `retryingClient` and export it non-defaultly?

File: package.json
Comment (grzpab) [inline package.json:1]: License check (no changes, the "red" libraries have been already discussed)
image

File: test/dowloadService.test.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		await downloadService.downloadFileIfNeeded(
+			`https://test.com/test`,
+			// @ts-expect-error passing a string instead of URI, because URI cannot be imported from vscode
+			'/',
+			'755',
+		);
+
+		expect(fs.writeFile).toBeCalledWith(
+			'/',
+			new Uint8Array([84, 101, 115, 116]),
```
Comment (grzpab) [inline test/dowloadService.test.ts:80]: what does that mean?

Comment (DmytroHryshyn) [reply test/dowloadService.test.ts:80]: We mock `get` response with `Test`, and expect that writeFile will be called with uint representation of that string. fs.writeFile is mocked, so we can check args it was called with

Comment (grzpab) [reply test/dowloadService.test.ts:80]: Ah ok, the array in question is basically `Test`:
```
new Uint8Array([84, 101, 115, 116]).forEach((x) => console.log(String.fromCharCode(x)))
```

**Review by grzpab — APPROVED** [review]

**Pull request merged**